clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1515,7 +1515,7 @@ public void handleResponse(AbstractResponse response) {
             } else if (error == Errors.COORDINATOR_NOT_AVAILABLE || error == Errors.NOT_COORDINATOR) {
                 lookupCoordinator(FindCoordinatorRequest.CoordinatorType.TRANSACTION, transactionalId);
                 reenqueue();
-            } else if (error == Errors.COORDINATOR_LOAD_IN_PROGRESS || error == Errors.CONCURRENT_TRANSACTIONS) {
+            } else if (error == Errors.CONCURRENT_TRANSACTIONS || error.exception() instanceof RetriableException) {
```

Comment:
I think `ConcurrentTransactionsException` should be retriable.

Reply:
It is not.

Reply:
Hmm, I don't think there are any dependencies outside `TransactionManager`. It seems like a safe change? It _is_ a retriable error: we just haven't marked it as such explicitly.